src/rc/seedrng.c

```diff
@@ -392,10 +433,27 @@ int main(int argc _unused, char *argv[] _unused)
 	struct timespec realtime = { 0 }, boottime = { 0 };
 	struct blake2s_state hash;
 
+	atexit(cleanup);
+	applet = basename_c(argv[0]);
+	if (geteuid())
+		eerrorx("%s: superuser access is required", applet);
 	umask(0077);
-	if (getuid())
-		eerrorx("This rc helper program requires root");
 
+	while ((opt = getopt_long(argc, argv, getoptstring,
```

Comment:
Sounds like it'd make more sense to either do:
- changes via environment variable; or
- changes via getopt_long,
but not both at the same time. Just pick one.

Reply:
I tend to agree with this, but there are several places in OpenRC here we do override environment variable settings with command line options, so I added the capability to have some form of consistency with those. If I change that, I'll probably just go with command line parsing.

Reply:
Going only with command line options seems like a good choice.

Is there a reason or a utility in keeping the environment variables around? Keep in mind that this utility lives in /rc, so it's not like it's generally callable by most things. Pretty isolated tool.

Reply:
Also, I assume you want to replace /etc/init.d/urandom's usage of the environment variables with the command line switches as part of this patchset?

Reply:
If we need to get this into a 0.44.x release, I'll not bother urandom too much before that (I'll rework it in a separate pr) so I can backport easily to 0.44.x and not surprise users.
The previous commit which adds seedrng, along with the squashed one from this pr  is what I would backport to 0.44.x. After that I want to turn urandom into a virtual service (using provide in the depend() function) which will be provided by /etc/init.d/seedrng on Linux and /etc/init.d/urandom-bsd on the bsds.
This may require people to add the appropriate service to their runlevels if they are upgrading.
If we don't need seedrng in 0.44.x, I can keep going in this pr.

Reply:
Probably seedrng should go into the next release, yea.